includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -2554,13 +2554,14 @@ public function get_level3_data_from_order( WC_Order $order ): array {
 		$process_item  = static function( $item ) use ( $currency ) {
 			// Check to see if it is a WC_Order_Item_Product or a WC_Order_Item_Fee.
 			if ( is_a( $item, 'WC_Order_Item_Product' ) ) {
-				$subtotal   = $item->get_subtotal();
-				$product_id = $item->get_variation_id()
+				$subtotal     = $item->get_subtotal();
+				$product_id   = $item->get_variation_id()
 					? $item->get_variation_id()
 					: $item->get_product_id();
+				$product_code = substr( $product_id, 0, 12 );
 			} else {
-				$subtotal   = $item->get_total();
-				$product_id = substr( sanitize_title( $item->get_name() ), 0, 12 );
+				$subtotal     = $item->get_total();
+				$product_code = substr( sanitize_title( $item->get_name() ), 0, 12 );
```

Comment:
To keep codes unique we can use tricks like converting them from decimal base to 36-character based e.g. with:
```php
base_convert(12345678901234567890, 10, 36)
```
Outputs `c5m8nq6iv3` which fits into 10 characters.

Reply:
Thanks for the review! I had opted base 36 earlier, it was suggested in the [solution in issue #4411 ](https://github.com/Automattic/woocommerce-payments/issues/4411#:~:text=Currently%2C%20there%20is%20a%20fix%20in%20place%20via%20this%20PR%20for%20WCCOM.)too. 

Favored truncation later, since @kalessil  advised keeping `product_id` as unchanged as possible. For [non product items](https://github.com/Automattic/woocommerce-payments/blob/de63c3322d99a9c91676bc8c61e3a69b916cd2b2/includes/class-wc-payment-gateway-wcpay.php#L2564) in the code above, and for zero cost products ( where order contains only shipping cost ) , we are anyway not giving unique `product_code` .

Reply:
@dmvrtx: I declined a similar proposal in a recent iteration in favor of truncating. Converting will make matching items even more complicated - it's easier to match items by first characters if they are truncated.

Reply:
Apologies but I need to reopen the conversation. The Stripe docs [here](https://stripe.com/docs/level3) define the `product_code` like this: 'Up to 12 characters that uniquely identify the product.'. My concern is that using truncate we cannot ensure that the `product_code` is unique. I'd rather use the same approach that @dmvrtx suggested, provided that we only use the converted product id for level3 data purposes. What do you think @kalessil @nagpai @dmvrtx

Reply:
As suggested by @kalessil  during our call, since this is not a high priority issue, should we go ahead with the truncation method for now and merge. Subsequently I will document this on an internal p2 post and revisit, should we change anything. I hope that works? - @mgascam , @dmvrtx  . Thanks

Reply:
@nagpai, yes, please go ahead and merge.